File: BOJ/10000/10828_스택.py
import sys
n = int(sys.stdin.readline())
stack = []

for i in range(n):
    op = sys.stdin.readline().strip()
    if "push" in op:
        op, x = op.split()
        stack.append(x)
    elif op == "pop":
        if(len(stack)):
            print(stack.pop())
        else:
            print(-1)
    elif op == "size":
        print(len(stack))
    elif op == "empty":
        if(len(stack)):
            print(0)
        else:
            print(1)
    else:
        if(len(stack)):
            print(stack[-1])
        else:
            print(-1)

'''
op = sys.stdin.readline().strip().split()

if op[0] == "push":
    stack.append(int(op[1]))
elif op[0] == "pop":
    if stack:
        print(stack.pop())
    else:
        print(-1)

'''

File: BOJ/10000/10828_스택.cpp
#include <iostream>
#define MAX 10001

using namespace std;

typedef struct Stack
{
    int stack[MAX];
    int size;
} Stack;

Stack s;

void push(int x)
{
    s.stack[s.size++] = x;
}

int pop()
{
    if (s.size <= 0)
        return -1;

    int item = s.stack[--s.size];
    s.stack[s.size] = 0;
    return item;
}

bool empty()
{
    return s.size == 0;
}

int top()
{
    if (s.size <= 0)
        return -1;

    return s.stack[s.size - 1];
}

int main(void)
{
    int n;
    cin >> n;

    string op;

    for (int i = 0; i < n; i++)
    {
        cin >> op;

        if (op == "push")
        {
            int x;
            cin >> x;
            push(x);
        }
        else if (op == "pop")
            cout << pop() << "\n";
        else if (op == "size")
            cout << s.size << "\n";
        else if (op == "empty")
            cout << empty() << "\n";
        else
            cout << top() << "\n";
    }
}